x/leverage/keeper/math.go

```diff
@@ -0,0 +1,16 @@
+package keeper
+
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
+// Interpolate takes a line defined by two points (xMin, yMin) and (xMax, yMax), then finds the y-value of the
+// point on that line for an input x-value. It will return yMin if xMin = xMax (i.e. a vertical line).
+// While this function is intended for interpolation (xMin < x < xMax), it works correctly even when x is outside
+// that range or when xMin > xMax.
+func (k Keeper) Interpolate(ctx sdk.Context, x, xMin, yMin, xMax, yMax sdk.Dec) sdk.Dec {
```

Comment:
Do we have a unit test for this function?

Reply:
Yes - it was in `keeper_test.go`, but I just separated it out into a new file `math_test.go`